src/vellum/workflows/utils/tests/test_functions.py

```diff
@@ -13,4 +13,31 @@ def my_function():
     # THEN it should return the compiled function definition
     assert compiled_function == FunctionDefinition(
         name="my_function",
+        parameters={"type": "object", "properties": {}, "required": []},
+    )
+
+
+def test_compile_function_definition__all_args():
```

Comment:
I considered `pytest.mark.parameterize`, but it was awkward to define the functions as a parameter. these tests are simple enough to do as individual ones anyway